### DISCUSSION TITLE
How are multiple API clients handled?

### DISCUSSION THREAD

**Discussion by geordie2020** [bentoml/OpenLLM#306] [General]
Hi, i'm investigating ways to serve LLMs to a small group of people. Therefore e.g. 10 clients/people would need to connect via a webinterface to the OpenLLM served API, and be able to query the LLM at the same time.

How does OpenLLM handle (potentially concurrent) multiple requests? Is there some queueing with round robin going on? 
Or maybe time multiplexing?

**Comment by aarnphm** [comment]
I have a branch that support continuous batching. For now it will use the vLLM backend, but yes we are working on supporting this for general usecase.

**Reply by aarnphm** [reply]
Right now we have some sort of naive batching, but it is not efficient enough.
